JustSaying.Models/Message.cs

```diff
@@ -8,11 +8,20 @@ protected Message()
         {
             TimeStamp = DateTime.UtcNow;
             Id = Guid.NewGuid();
+            CorrelationId = Id;
```

Comment:
There's also a related issue in that I don't think JustSaying has a notion of system/header fields when serializing/deserializing. If the Message base class has a property added with the same name as field  being used by an application's message class already, there might be compilation errors (if access modifiers or types differ) or overwritten data. Just adding a field to the base class therefore becomes a potentially breaking change unless we can be confident the name is not in use already.

Ideally it would be nice if JustSaying wrapped app-level message classes in an Envelope<T> instead of requiring inheritance of a base class -  not entirely sure how feasible/difficult it would be to retro fit something like that.